ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala

```diff
@@ -979,16 +974,15 @@ abstract class AbstractWebsocketServiceIntegrationTest
       query = s"""[{"templateIds": ["Iou:Iou"]}]"""
       streamError <- singleClientQueryStream(jwt, uri, query, Some(offsetBeforeArchive))
         .runWith(Sink.seq)
-        .failed
-    } yield inside(streamError) { case t: PeerClosedConnectionException =>
-      // TODO #13506 descriptive/structured error.  The logs when running this
-      // test include
-      //     Websocket handler failed with FAILED_PRECONDITION: PARTICIPANT_PRUNED_DATA_ACCESSED(9,0):
-      //     Transactions request from 0000000000000006 to 0000000000000008
-      //     precedes pruned offset 0000000000000007
-      // but this doesn't propagate to the client
-      t.closeCode should ===(1011) // see RFC 6455
-      t.closeReason should ===("internal error")
+    } yield inside(streamError) { case Seq(TextMessage.Strict(text)) =>
+      val errorResponse = decodeErrorResponse(text)
+      errorResponse.status shouldBe StatusCodes.BadRequest
+      errorResponse.ledgerApiError
+        .map { it =>
+          it.code shouldBe 9
+          it.message should include("PARTICIPANT_PRUNED_DATA_ACCESSED")
+        }
+        .getOrElse(fail("ledger api error was expected to be included in the response"))
```

Comment:
I do not think this should be a message on the stream with a successful close; it should be a stream failure as it was before.